raco/language/myrialang.py

```diff
@@ -806,6 +838,9 @@ def __repr__(self):
     def num_tuples(self):
         return self._num_tuples
 
+    def partitioning(self):
```

Comment:
> Or maybe I'm missing something and all the partitioning info is propagated up the query tree and relevant optimizations are performed before any expressions are optimized into MyriaQueryScan.

No, your first interpretation is correct. `partitioning` is functional like `num_tuples`, so what matters is the structure of the AST at the moment you call `MyriaStore.partitioning()` when you are compiling it to json.